shub/image/list.py

```diff
@@ -140,12 +143,17 @@ def raise_shub_image_info_error(error):
     if not isinstance(spiders_list, list):
         raise_shub_image_info_error('spiders section must be a list')
 
+    project_type = metadata.get('project_type')
```

Comment:
Validating the variable type is different from validating the variable value. IMO it is the client's responsibility to not make malformed requests (and the server's responsibility to reject correctly formed but disallowed values). Checking the type of `project_type` would be consistent with how we handle `spiders` and its entries (checking that the type is list or string, respectively).

More importantly, if we allow `None` here then `shub image test` will pass for an image that breaches the contract.